client/src/components/Listings/ProfileList/ProfileList.tsx

```diff
@@ -1,35 +1,55 @@
-import { useState } from 'react';
-import { Box, Grid, Grow, Typography } from '@material-ui/core';
+import { useState, useEffect } from 'react';
+import { Box, Button, Grid, Grow, Typography } from '@material-ui/core';
+import Pagination from '@material-ui/lab/Pagination';
+import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';
+import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
 
 import useStyles from './useStyles';
 import { useAuth } from '../../../context/useAuthContext';
 import { useSnackBar } from '../../../context/useSnackbarContext';
+import { Profile } from '../../../interface/Profile';
 import ProfileCard from '../ProfileCard/ProfileCard';
 import LoadingSpinner from '../../LoadingSpinner/LoadingSpinner';
-import Pagination from '@material-ui/lab/Pagination';
 
 export default function ProfileList(): JSX.Element {
   const { updateSnackBarMessage } = useSnackBar();
-  const { sitterProfiles, loading, errorMsg } = useAuth();
+  const { sortByPrice, loading, errorMsg } = useAuth();
   const [page, setPage] = useState<number>(1);
+  const [sortAscPrice, setSortAscPrice] = useState<boolean>(true);
+  const [sortedProfiles, setSortedProfiles] = useState<Profile[]>([]);
   const itemsPerPage = 9;
-  const noOfPages = Math.ceil(sitterProfiles.length / itemsPerPage);
+  const noOfPages = Math.ceil(sortedProfiles.length / itemsPerPage);
   const classes = useStyles();
 
+  useEffect(() => {
+    const profiles = sortByPrice(sortAscPrice);
+    setSortedProfiles(profiles);
+  }, [sortAscPrice, sortByPrice]);
+
```

Comment:
- To improve on attention to details when it comes to make sure you are outputting quality code to the main branch, test all base cases and edge cases before you merge your code in. Small changes could have a rippling effect on other features.